internal/cli/cmd/cluster/buildx.go

```diff
@@ -246,7 +305,40 @@ func newCleanupBuildxCommand() *cobra.Command {
 		}
 
 		return withStore(dockerCli, func(txn *store.Txn) error {
-			return txn.Remove(*name)
+			state, err := ensureStateDir(*stateDir, buildkitProxyPath)
+			if err != nil {
+				return err
+			}
+
+			if !proxyAlreadyExists(state) {
+				console.SetStickyContent(ctx, "build", "State file not found. Nothing to cleanup.")
+				return nil
+			}
+
+			var md buildxMetadata
+			if err := files.ReadJson(filepath.Join(state, metadataFile), &md); err != nil {
+				return err
+			}
+
+			for _, inst := range md.Instances {
+				if inst.BackgroundProxyPID > 0 {
+					process, err := os.FindProcess(inst.BackgroundProxyPID)
+					if err != nil {
+						return err
+					}
+
+					err = process.Signal(syscall.SIGTERM)
+					if err != nil && !errors.Is(err, os.ErrProcessDone) {
```

Comment:
Avoid re-assignments:

`if err := process.Signal(syscall.SIGTERM); err != nil && !errors.Is(err, ...) {`